node/readline/promises.ts

```diff
@@ -0,0 +1,194 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+import { Readline } from "../internal/readline/promises.mjs";
+
+import {
+  Interface as _Interface,
+  kQuestion,
+  kQuestionCancel,
+} from "../internal/readline/interface.mjs";
+import { AbortError } from "../internal/errors.ts";
+import { validateAbortSignal } from "../internal/validators.mjs";
+
+import { kEmptyObject } from "../internal/util.mjs";
+import type { Abortable } from "../_events.d.ts";
+
+// NOTE: These type redefinitions are intended to avoid circular dependencies.
+// import type {
+//   AsyncCompleter,
+//   Completer,
+//   ReadLineOptions,
+// } from "../_readline.d.ts";
+type Completer = (line: string) => CompleterResult;
+type AsyncCompleter = (
+  line: string,
+  callback: (err?: null | Error, result?: CompleterResult) => void,
+) => void;
+type CompleterResult = [string[], string];
+interface ReadLineOptions {
+  input: ReadableStream;
+  output?: WritableStream | undefined;
+  completer?: Completer | AsyncCompleter | undefined;
+  terminal?: boolean | undefined;
+  history?: string[] | undefined;
+  historySize?: number | undefined;
+  prompt?: string | undefined;
+  crlfDelay?: number | undefined;
+  removeHistoryDuplicates?: boolean | undefined;
+  escapeCodeTimeout?: number | undefined;
+  tabSize?: number | undefined;
```

Comment:
Do these need to be both optional and explicitly allow `undefined`?

Reply:
I had just copy-pasted the contents of _readline.d.ts.
As you suggested, I have placed these in a separate file. I left the `undefined` in that file as well, wondering if it would be better to keep the contents of the original forked file as much as possible.